### PR TITLE
HDDS-2499. IsLeader information is lost when update pipeline state.

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/HDDS-2499?filter=12346871

### CODE REVIEW

**Comment by ChenSammi** [comment]
You are right. I'm asking @swagle for the review.  Sorry to ping the wrong person.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestSCMPipelineManager.java [hunk 1]
```diff
@@ -174,18 +174,17 @@ public void testPipelineReport() throws IOException {
...
-    for (DatanodeDetails dn: pipeline.getNodes()) {
-      PipelineReportFromDatanode pipelineReportFromDatanode =
-          TestUtils.getPipelineReportFromDatanode(dn, pipeline.getId());
-      // pipeline is not healthy until all dns report
-      Assert.assertFalse(
-          pipelineManager.getPipeline(pipeline.getId()).isHealthy());
-      pipelineReportHandler
-          .onMessage(pipelineReportFromDatanode, new EventQueue());
-    }
+    List<DatanodeDetails> nodes = pipeline.getNodes();
```
Comment (xiaoyuyao) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestSCMPipelineManager.java]: There is a similar negative case in Line 199-205. It does not cause failure but we can fix them similarly. The previous approach can also be fixed with slightly modification of TestUtils.getPipelineReportFromDatanode as demoed in the dup PR #195.

**Comment by xiaoyuyao** [comment]
Thanks for the update. LGTM, +1.

**Comment by ChenSammi** [comment]
Thanks @xiaoyuyao  and @swagle for the review.

**Pull request merged**